collections/src/main/scala/strawman/collection/ArrayOps.scala

```diff
@@ -1,34 +1,47 @@
 package strawman
 package collection
 
-import scala.{AnyVal, Array, ArrayIndexOutOfBoundsException, Char, Int, throws}
+import scala.{AnyVal, Array, ArrayIndexOutOfBoundsException, Int, throws, Boolean, Serializable, Unit, `inline`}
 import scala.Predef.???
 import mutable.{ArrayBuffer, GrowableBuilder}
-
+import immutable.ImmutableArray
 import scala.reflect.ClassTag
 
-class ArrayOps[A](val xs: Array[A])
-  extends AnyVal
-    with IterableOnce[A]
-    with IndexedSeqOps[A, immutable.IndexedSeq, Array[A]]
-    with StrictOptimizedIterableOps[A, Seq, Array[A]]
-    with ArrayLike[A] {
+object ArrayOps {
+  class WithFilter[A](p: A => Boolean, ao: ArrayOps[A]) extends collection.WithFilter[A, immutable.IndexedSeq] {
+    protected[this] def filtered = View.Filter(ao.toIterable, p, isFlipped = false)
+    def map[B](f: A => B): immutable.IndexedSeq[B] = ao.iterableFactory.from(View.Map(filtered, f))
+    def flatMap[B](f: A => IterableOnce[B]): immutable.IndexedSeq[B] = ao.iterableFactory.from(View.FlatMap(filtered, f))
+    def foreach[U](f: A => U): Unit = filtered.foreach(f)
+    def map[B: ClassTag](f: A => B): Array[B] = ao.fromTaggedIterable(View.Map(filtered, f))
+    def flatMap[B: ClassTag](f: A => IterableOnce[B]): Array[B] = ao.fromTaggedIterable(View.FlatMap(filtered, f))
+    def withFilter(q: A => Boolean): WithFilter[A] = new WithFilter[A](a => p(a) && q(a), ao)
+  }
+}
+
+class ArrayOps[A](val xs: Array[A]) extends AnyVal
+  with IterableOnce[A]
+  with IndexedSeqOps[A, immutable.IndexedSeq, Array[A]]
+  with StrictOptimizedIterableOps[A, Seq, Array[A]]
+  with ArrayLike[A] {
+
+  protected def fromTaggedIterable[B: ClassTag](coll: Iterable[B]): Array[B] = coll.toArray[B]
+
+  // Note: Declaring the return type as ArrayOps.WithFilter[A] breaks Dotty 0.4.0-RC1
```

Comment:
Does that mean that we won’t get back an `Array` but an `IndexedSeq` if we call `map`?

Reply:
Could you file an issue in the dotty repository (lampepfl/dotty)? These issues have high priority and might be fixed in a few days.

Reply:
Strangely enough I was unable to reproduce the error again. Let's see how CI fares.